src/main/java/Controller/uploadSongController.java

```diff
@@ -1,11 +1,25 @@
 package Controller;
 import InputBoundary.uploadSongInputBoundary;
+import InputData.uploadSongInputData;
+import OutputBoundary.uploadSongOutputBoundary;
+import UseCase.UploadSongInteractor;
 
 public class uploadSongController {
 
-    uploadSongInputBoundary uploadSongInteractor;
-    public void upload(uploadSongInputBoundary uploadSongInteractor){
-        //TODO
+    uploadSongOutputBoundary uploadSongPresenter;
```

Comment:
we should decide whether we're actually still passing the presenter into the controller so that it can construct the controller -- i don't think you need to do anything about it for now, but just marking it for future discussion as this has come up on other PRs.